libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,209 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solRequire(_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (auto current = begin(_code); current != end(_code); ++current)
+	{
+		auto const& item = m_items.emplace_back(createAssemblyItemFromJSON(*current, m_sourceList));
+		if (item == Instruction::JUMPDEST)
+			solThrow(AssemblyImportException, "JUMPDEST instruction without a tag");
+		else if (item.type() == AssemblyItemType::Tag)
+			solRequire(
+				++current != end(_code) &&
+					createAssemblyItemFromJSON(*current, m_sourceList) == Instruction::JUMPDEST,
+				AssemblyImportException,
+				"JUMPDEST expected after tag."
+			);
+	}
+}
+
+AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json, std::vector<std::string> const& _sourceList)
+{
+	solRequire(_json.isObject(), AssemblyImportException, "Supplied JSON is not an object.");
+	static set<string> const validMembers{"name", "begin", "end", "source", "value", "modifierDepth", "jumpType"};
+	for (auto const& member: _json.getMemberNames())
+		solRequire(
+			validMembers.count(member),
```

Comment:
```suggestion
			validMembers.count(member) != 0,
```

Reply:
I leave it like it this - as also @ekpyron said - we use that pattern in many places.